docs/modules/processor/examples/test.js

```diff
@@ -0,0 +1,53 @@
+const asciidoctor = require('asciidoctor.js')();
+
+// tag::load-file[]
+var doc = asciidoctor.loadFile('sample.adoc');
+// end::load-file[]
+
+// tag::document-title[]
+console.log(doc.getDocumentTitle()); // The Documentation Chronicles: Based on True Events
```

Comment:
@mojavelinux It would be awesome if the expected result was generated dynamically. We could even create a custom include directive to read the expected value from a test case.

**code**
```js
describe('Extract information from a document', () => {
  it('should get the document title', () => {
    const doc = asciidoctor.loadFile('sample.adoc');
    // tag::document-title[]
    expect(doc.getDocumentTitle()).toBe('The Documentation Chronicles: Based on True Events');
    // end::document-title[]
  });
});
```

**documentation**
```
include-js-expect::{examplesdir}/spec.js[tags=document-title]
```

**result**
```js
console.log(doc.getDocumentTitle()); // The Documentation Chronicles: Based on True Events
```

* `doc.getDocumentTitle()` is extracted from `expect(${code})`
* `The Documentation Chronicles: Based on True Events` is extracted from `toBe(${expectedValue})`
* Then we apply a template `console.log(${code}); // ${expectedValue}`

Reply:
It should be possible with an include processor but I don't want to handle partial include (see https://github.com/asciidoctor/asciidoctor-extensions-lab/blob/master/lib/whitespace-include-processor.rb)

Reply:
@mojavelinux Do you think it could be a good idea to introduce an include content processor triggered once the content has been read ?

```rb
Extensions.register do
  include_content_processor do
    process do |doc, reader, source, attributes|
      reader.push_include source.gsub("\t", '    '), target, target, 1, attributes
      reader
    end

    def handles? target
      target.start_with? 'code/'
    end
  end
end
```